tests/31sync/06state.pl

```diff
@@ -1,4 +1,30 @@
-test "State is included in the initial sync",
+use Future::Utils qw( repeat );
+
+# call /sync repeatedly until it returns a result
+# with an event in the given room
+sub wait_for_event_in_room {
```

Comment:
Did you want to be using `await_event_for()` here instead? I'm aware that waits for an event to arrive on the event stream, rather than polling for some amount of state, but would that suffice somehow?